server_src/services/database/server-settings.service.js

```diff
@@ -0,0 +1,294 @@
+const ServerSettingsDB = require("../../models/ServerSettings.js");
+
+const { isEmpty } = require("lodash");
+
+class ServerSettings {
```

Comment:
Personally I'd call this `ServerSettingRepository` or `ServerSettingService` to be able to discern it from the database level (Schema's, low-level) as well as API level (REST validation, high level).  

To explain the difference I have in mind:
1) The Repository is a tool to wrap/extend/abstract/secure **storage** sources like file or database
2) the Service is a tool to combine underlying **data sources** so file/db/cache/middleware/external-api's/etc. The service is higher up and will do less low-level work like f.e. checking whether the values provided match our constraints (validation). It has little to do with the storage level (abstraction).

Therefore when I would've coded this, I'd call this a ServerSettingsService - it combines a cache with another storage (db). Secondly, I would move certain functionality to the cache to make it more useful and I would've created a ServerSettingsRepository class to extend the database work like explicit validation in order to make it much easier to get rid of MongoDB dependency (decoupling).

Let me know if you like this idea. I can also work it out in drawings to make it clearer.

Reply:
Oh btw, V2 and .NET based servers seem to do this pattern. 
- (Module) - hierarchical separation of code (registers Controller, Service, Repository, Entity and others)
- Controller - API and MVC request/response actions
- Service - Data source functionality (used by Controller)
- Repository - Data storage functionality (used by Service)
- Entity - Data modeling (used by Repository)

Some other constructs: 
- Manager - a entity manager responsible for 1 or multiple entities at the same time (used by Repository)
- Cache - special kind of storage controller in-memory
- Config - compact configuration model and manager (bit special as configuration is usually a startup flow where are merge action happens from terminal
- Constants/Data - constant data which define our application 

Note that you've introduced a PatreonsCache, but it contains no logic. I'd name that file PatreonsData or add it to the file with AppConstants

Reply:
I was already planning too name it this aha, I do clean up last. Maybe I should save you typing on a draft pr and do clean up first XD. 

The patreons cache isn't technically anything to do with the application to me. It's a separate list. I'm thinking of putting it on the OF website and making OF call that json list from there. I'd just write the file locally so the EJS has something to dump on settings. I want a fallback though for air gapped farms.